apidef/oas/default.go

```diff
@@ -98,6 +98,13 @@ func (s *OAS) BuildDefaultTykExtension(overRideValues TykExtensionConfigParams,
 		}
 
 		upstreamURL = s.Servers[0].URL
+		if s.isURLParamatrized(upstreamURL) {
+			for name, variable := range s.Servers[0].Variables {
+				if strings.Contains(upstreamURL, "{"+name+"}") {
+					upstreamURL = strings.ReplaceAll(upstreamURL, "{"+name+"}", variable.Default)
```

Comment:
we're not validating whether there is a default value for the variable.
api should return error when default value is provided for a variable. 
also can we refactor this variable replacement logic into a separate function?